api/modules/currency/views.py

```diff
@@ -33,3 +35,51 @@ def get_currency_exchange_rate(request, source_currency_code, target_currency_co
         return Response(exception_message, status=status.HTTP_503_SERVICE_UNAVAILABLE)
 
     return Response(response.to_json())
+
+
+@api_view(['GET'])
+def get_all_currency_exchange_rate(request, start_date, end_date, source_currency_code, target_currency_code):
+    """
+    Return currency exchange rates list between 2 dates
+    :param request:
+    :param start date:
+    :param end date:
+    :param source_currency_code:
+    :param target_currency_code:
+    :return 400 response if dates are incorrect
+    :return: 503 if Free Currency Converter api fails
+    :return: 200 successful
+    """
+    currency_list = []
+    y_start = start_date.split('-')[0]
+    m_start = start_date.split('-')[1]
+    d_start = start_date.split('-')[2]
+    start = datetime.datetime.strptime(y_start+m_start+d_start, '%d%m%Y').date()
+    y_end = end_date.split('-')[0]
+    m_end = end_date.split('-')[1]
+    d_end = end_date.split('-')[2]
+    end = datetime.datetime.strptime(y_end+m_end+d_end, '%d%m%Y').date()
+    if end < start:
+        error_message = "End Date is before Start Date"
+        return Response(error_message, status=status.HTTP_400_BAD_REQUEST)
+    try:
+        api_response_start = requests.get(CURRENCY_VALUE_DATE_API_URL.format(
+            start_date, source_currency_code, target_currency_code))
+        api_response_end = requests.get(CURRENCY_VALUE_DATE_API_URL.format(
+            end_date, source_currency_code, target_currency_code))
+        if not api_response_start.ok or not api_response_end.ok:
+            error_message = "Incorrect parameters please check dates"
+            return Response(error_message, status=status.HTTP_400_BAD_REQUEST)
+        delta = end - start
+        for day in range(delta.days + 1):
+            actual_date = str(start + timedelta(day))
+            response = requests.get(CURRENCY_VALUE_DATE_API_URL.format(
+                actual_date, source_currency_code, target_currency_code))
+            currency_value = response.json()
+            currency_list.append(CurrencyDate(date=str(actual_date),
```

Comment:
No need of passing date information
If there are 30 days, a list of 30 should mean currency rates are in order